feat: add feeds for the blog and vulnerabilities

rowanmanning opened the pull request:
This adds two new feeds to the site: `feed.xml` includes all blog posts, and `vulnerabilities.xml` includes all posts tagged with `vulnerabilities`.

You can validate the feeds via the [W3C Feed Validation Service](https://validator.w3.org/feed/). I also tried subscribing via Feedbin and all the posts appear as expected.

Resolves #1763

<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

rowanmanning committed:
feat: add feeds for the blog and vulnerabilities

This adds two new feeds to the site: `feed.xml` includes all blog posts,
and `vulnerabilities.xml` includes all posts tagged with
`vulnerabilities`.

Resolves #1763

rowanmanning commented on _includes/feed-entry.xml:
```diff
@@ -0,0 +1,17 @@
+{%- assign entry = include.entry -%}
+<entry>
+  <id>{{ site.url }}{{ entry.url }}</id>
```
In Atom, the `id` is [expected to be an IRI](https://www.ietf.org/rfc/rfc4287.html#section-4.2.6) however using the page URL seems to be relatively common practice. An alternative would be to mimic [GitHub's releases feed](https://github.com/expressjs/express/releases.atom) and use something like

```
tag:expressjs.com,<YEAR>:https://expressjs.com/<POST_URL>
```

however this doesn't get around the fact that the ID is based on the page URL.

bjohansebas replied:
I like either of those two options, and i think both are fine.

rowanmanning replied:
OK let's just go with the URL 👍

rowanmanning commented on _includes/feed-entry.xml:
```diff
@@ -0,0 +1,17 @@
+  <published>{{ entry.date | date_to_xmlschema }}</published>
+  <updated>{{ entry.last_modified_at | default: post.date | date_to_xmlschema }}</updated>
+  {%- for tag in post.tags %}
+  <category term="{{ tag | replace: ',', '' | xml_escape }}" />
```
This is overcomplicated to fix an issue with the current posts. [Jekyll post tags](https://jekyllrb.com/docs/posts/#tags-and-categories) are supposed to be space-separated. The tags in the current posts are comma-separated which results in trailing commas appearing in the tags.

I wasn't sure whether to add this little hack or to edit all of the posts to correct the tags. If the latter than I'll be setting the last modified date of each post; happy to do this if you want me to.

bjohansebas replied:
It's really better to fix it at the source than to make small hacks

rowanmanning replied:
Great 👍 fixed in a separate commit

rowanmanning commented on _layouts/feed.xml:
```diff
@@ -0,0 +1,10 @@
+  <link href="{{ site.url }}{{ page.html_url }}" rel="alternate" type="text/html" />
+  <link href="{{ site.url }}{{ page.url }}" rel="self" type="application/atom+xml" />
+  <id>{{ site.url }}{{ page.url }}</id>
+  <rights>Copyright © {{ site.time | date: '%Y' }}, The Express Contributors</rights>
```
Is this copyright OK? I made assumptions based on the license. It's not a required element so I can ditch it or make changes.

bjohansebas replied:
I don't think we need to include the date — we don't have it in the license of the repo.

Ping @expressjs/express-tc in case they have an opinion on this.

rowanmanning replied:
I removed the date. I'm unable to @ the team though, are you able to give them a heads up?

wesleytodd replied:
Hm, I don't know much about how to represent this in feeds, but this is the correct guidance to follow if we need to have this at all: https://github.com/openjs-foundation/cross-project-council/blob/main/governance/IP_POLICY_GUIDANCE.md#2-copyright-notices

rowanmanning replied:
Ah great, that's all the guidance I need, thanks. I've copied the format exactly from the linked doc 👍

bjohansebas reviewed (commented):
Overall, everything looks good, @rowanmanning  thanks for working on this.

rowanmanning committed:
fix: use space-delimited tags

Jekyll expects tags to be space-delimited, using commas results in the
tags themselves containing a comma. See docs:
https://jekyllrb.com/docs/posts/#tags

Fixing this allows us to stop working around the trailing commas.

wesleytodd reviewed (approved):
I think this is great! Thanks!

bjohansebas reviewed (approved):
LGTM!

bjohansebas committed:
Merge branch 'gh-pages' into blog-feed

bjohansebas commented:
@rowanmanning Thanks!
